src/models/user/user.types.ts

```diff
@@ -12,5 +11,25 @@ export interface IUser {
   updatedAt: Date;
 }
 
+export interface IUserRegistrationDetails {
+  email: string;
+  firstName: string;
+  lastName: string;
+  plainTextPassword: string;
+}
+
 export interface IUserDocument extends IUser, Document {}
-export interface IUserModel extends Model<IUserDocument> {}
+export interface IUserModel extends Model<IUserDocument> {
+  findOneByEmail: ({ email }: { email: string }) => Promise<IUserDocument>;
+  createUser: ({
+    email,
+    firstName,
+    lastName,
+    plainTextPassword,
+  }: {
+    email: string;
+    firstName: string;
+    lastName: string;
+    plainTextPassword: string;
+  }) => Promise<IUserDocument>;
+}
```

Comment:
Since I defined a model controller (function) on the `UserModel`, this allows Typescript to play nicely with Mongoose